xml/admin_ds_backup.xml

```diff
@@ -140,5 +142,92 @@
     </para>
    </listitem>
   </itemizedlist>
+ </sect2>
+ </sect1>
+ <sect1 xml:id="restore-ceph">
+  <title>Restoring a Ceph Node</title>
+  <para>
+   The basic procedure for recovering a node from backup is to reinstall the
+   node, replace its configuration files, and then re-orchestrate the cluster
+   so that the replacement node is re-added. 
+  </para>
+  <orderedlist>
+   <listitem>
+    <para>
+     Re-install the node. (See <link
+     xlink:href="https://documentation.suse.com/ses/6/single-html/ses-deployment/#ceph-install-stack"/>
+     steps 1-4.)
+    </para>
+   </listitem>
+   <listitem>
+    <para>
+     Update to the latest patch level (See: <link
+      xlink:href="https://documentation.suse.com/sles/15-SP1/single-html/SLES-admin/#sec-zypper-softup-update"/>.)
+    </para>
+   </listitem>
+   <listitem>
+    <para>
+     Install &salt; and &deepsea;. (See <link
+      xlink:href="https://documentation.suse.com/ses/6/single-html/ses-deployment/#ceph-install-stack"/>
+     steps 5-7.)
+    </para>
+   </listitem>
+   <listitem>
+    <para>
+     Restore the old, saved &deepsea; and &salt; configuration files.
+    </para>
+    <para>
+     It may be necessary to set the permissions appropriately on the replaced
+     files and manually enable services. For instance:
+<screen>&prompt.smaster; chown salt:root /etc/salt/pki/master/master.pem
+&prompt.smaster; chown salt:root /etc/salt/pki/master/master.pub
+&prompt.smaster; chown -R salt:salt /srv/pillar/ceph/
+&prompt.smaster; systemctl enable salt-api
+&prompt.smaster; systemctl start salt-api
+&prompt.smaster; systemctl enable salt-minion
+&prompt.smaster; systemctl start salt-minion
+&prompt.smaster; systemctl enable salt-master
+&prompt.smaster; systemctl start salt-master</screen>     
+    </para>
+   </listitem>
+   <listitem>
+    <para>
+     Register the minions against the new &salt; master. (See <link
+      xlink:href="https://documentation.suse.com/ses/6/single-html/ses-deployment/#ceph-install-stack"/>
+     steps 8-11.)
+    </para>
+   </listitem>
+   <listitem>
+    <para>
+     Restore anything that was local to the admin node.
+    </para>
+   </listitem>
+   <listitem>
+    <para>
+     Run stages 0-2. (See <link
+      xlink:href="https://documentation.suse.com/ses/6/single-html/ses-deployment/#ds-depl-stages"/>
+     steps 1-7.)
+    </para>
+   </listitem>
+   <listitem>
+    <para>
+     Make sure grains and pillar data are set.
```

Comment:
How can we verify the grains and pillar data?

Reply:
I don't know. I welcome suggestions who to ask.

Reply:
Generically, the commands `salt '*' pillar.items` and `salt '*' grains.items` will dump everything for every minion.  A more specific command such as `salt '*' pillar.get roles` and `salt '*' grains.get ceph` might be more useful and less noisy.

Reply:
> Generically, the commands `salt '*' pillar.items` and `salt '*' grains.items` will dump everything for every minion. A more specific command such as `salt '*' pillar.get roles` and `salt '*' grains.get ceph` might be more useful and less noisy.

I am afraid I am not familiar enough with using Salt to parse this. Are these commands to be run _before_ the backup, to generate files to back up, or are they commands to run _after_ it to verify that some settings are correct?

Reply:
Oh, I meant that either set of these commands might be suitable as an example directly under "Make sure grains and pillar data are set".  In other words, it's simply a verification that you can run two queries and get some results.

Since we are dealing with a site's own backup, the values will be different.  There's probably a tactful way to ask the admin "does the output look correct to you?".  For instance, if you went with the "roles" query, then the expectation is that all the minions have the assigned roles before loss of the Salt master.

Reply:
Thank you! Fixed.